File: js-sdk/src/timelineManager.js
class FyipeTimelineManager {
    constructor(options) {
        this.options = options;
        this.timeLineStack = [];
    }
    _addItemToTimeline(item) {
        // get the size of the stack
        if (this.timeLineStack.length === this.options.maxTimeline) {
            // this.timeLineStack.shift(); // remove the oldest item
            return; // It discards new timline update once maximum is reached
        }
        // add time to it
        item.timestamp = Date.now();
        // add a new item to the stack
        this.timeLineStack.push(item);
        return true;
    }
    addToTimeline(item) {
        this._addItemToTimeline(item);
    }
    // return the timeline
    getTimeline() {
        return this.timeLineStack;
    }
    // clear the timeline
    clearTimeline() {
        this.timeLineStack = [];
    }
}
export default FyipeTimelineManager;

File: ruby-sdk/fyipe/lib/fyipe/timelineManager.rb
class FyipeTimelineManager
    def initialize(options)
        @options = options
        @timelineStack = []
    end

    def addToTimeline(item)
        addItemToTimeline(item)
    end
    
    # return the timeline
    def getTimeline()
        return @timeLineStack
    end
    
    # clear the timeline
    def clearTimeline()
        @timeLineStack = []
    end 

    private_class_method def addItemToTimeline(item)
        # get the size of the stack
        if (@options['maxTimeline'] != nil and (@timeLineStack.length() == @options['maxTimeline']))
            return # It discards new timeline update once maximum is reached
        end

        # add time to it
        # current date and time
        time = Time.now
        now = time.inspect

        item["timestamp"] = now
        
        # add a new item to the stack
        @timeLineStack.push(item)
        return true
    end
end 